cpp/cmake/thirdparty/patches/nanoarrow_override.json

```diff
@@ -0,0 +1,18 @@
+
+{
+  "packages" : {
+    "nanoarrow" : {
+      "version" : "0.7.0.dev",
+      "git_url" : "https://github.com/apache/arrow-nanoarrow.git",
+      "git_tag" : "4bf5a9322626e95e3717e43de7616c0a256179eb",
+      "git_shallow" : false,
+      "patches" : [
+        {
+          "file" : "${current_json_dir}/nanoarrow_clang_tidy_compliance.diff",
```

Comment:
Is this patch still relevant? I don’t see it here. Can this override be removed?

Reply:
@zeroshade Can we remove this? I'm okay with merging this PR and opening a follow-up for cleanup, since CI has passed.

Reply:
Looks like #17618 removed this patch. I'll merge this PR and file a follow-up to delete this override.

Reply:
I realized we are missing some reviews. If we need to push any changes to this branch for other reviews, we should go ahead and delete this. Otherwise I think merging this and removing the overrides in a follow-up is the best case scenario.